contrib/verifybinaries/README.md

```diff
@@ -1,30 +1,77 @@
+decide which of these keys you would like to trust. For each key you want to trust, you
+must obtain that key for your local GPG installation.
 
-The script returns 0 if everything passes the checks. It returns 1 if either the signature check or the hash check doesn't pass. If an error occurs the return value is 2.
+You can obtain these keys by
+  - through a browser using a key server (e.g. keyserver.ubuntu.com),
+  - manually using the `gpg --keyserver <url> --recv-keys <key>` command, or
+  - you can run the packaged `verifybinaries.py ... --import-keys` script to
+    have it automatically retrieve unrecognized keys.
 
+#### Usage
 
+This script attempts to download the checksum file (`SHA256SUMS`) and corresponding
+signature file `SHA256SUMS.asc` from https://bitcoincore.org and https://bitcoin.org.
+
+It first checks if the checksum file is valid based upon a plurality of signatures, and
+then downloads the release files specified in the checksum file, and checks if the
+hashes of the release files are as expected.
+
+If we encounter pubkeys in the signature file that we do not recognize, the script
+can prompt the user as to whether they'd like to download the pubkeys. To enable
+this behavior, use the `--import-keys` flag.
+
+The script returns 0 if everything passes the checks. It returns 1 if either the
+signature check or the hash check doesn't pass. An exit code of >2 indicates an error.
+
+See the `Config` object for various options.
+
+#### Examples
+
+Validate releases with default settings:
+```sh
+./contrib/verifybinaries/verify.py 22.0
+./contrib/verifybinaries/verify.py 22.0-rc2
+./contrib/verifybinaries/verify.py bitcoin-core-23.0
+./contrib/verifybinaries/verify.py bitcoin-core-23.0-rc1
+```
+
+Get JSON output and don't prompt for user input (no auto key import):
+
+```sh
+./contrib/verifybinaries/verify.py 22.0-x86 --json
+```
+
+Don't trust builder-keys by default, and rely only on local GPG state and manually
+specified keys, while requiring a threshold of at least 10 trusted signatures:
 ```sh
-./verify.py bitcoin-core-0.11.2
-./verify.py bitcoin-core-0.12.0
-./verify.py bitcoin-core-0.13.0-rc3
+./contrib/verifybinaries/verify.py 22.0-x86 \
+    --no-trust-builder-keys \
+    --trusted-keys 74E2DEF5D77260B98BC19438099BAD163C70FBFA,9D3CC86A72F8494342EA5FD10A41BDC3F4FAFF1C \
+    --min-trusted-sigs 10
 ```
 
 If you only want to download the binaries of certain platform, add the corresponding suffix, e.g.:
 
 ```sh
-./verify.py bitcoin-core-0.11.2-osx
-./verify.py 0.12.0-linux
-./verify.py bitcoin-core-0.13.0-rc3-win64
+./contrib/verifybinaries/verify.py bitcoin-core-22.0-osx
+./contrib/verifybinaries/verify.py bitcoin-core-22.0-rc2-win64
 ```
 
 If you do not want to keep the downloaded binaries, specify anything as the second parameter.
 
 ```sh
-./verify.py bitcoin-core-0.13.0 delete
+./contrib/verifybinaries/verify.py bitcoin-core-22.0 delete
```

Comment:
From https://github.com/bitcoin/bitcoin/pull/23020#discussion_r1117050954
> Should we just add an explicit `--delete` argument to the script, rather than "append a random anything", as was proposed in https://github.com/bitcoin/bitcoin/pull/26985.

Reply:
Looks like this has been implemented as `pub --cleanup`. A quick test shows it working as intended. Will update the doc.